fhirAssembly/src/test/java/io/vantiq/extsrc/fhirAssembly/TestFhirUseAccessToken.java

```diff
@@ -0,0 +1,304 @@
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "Basic",
+                                        "basicAccessToken", tokenValue);
+        doOperations("Bearer", tokenValue, config, true);
+    }
+    
+    @Test
+    public void testOperationsWithAppManagedConfigBasedTokenAndRealm() {
+        String tokenValue = "ImALittleTeaPot";
+        String tokenRealm = "HonaleeLandOfPuff";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "ApplicationManaged",
+                                        "basicRealm", tokenRealm,
+                                        "basicAccessToken", tokenValue);
+        doOperations(tokenRealm, tokenValue, config, true);
+    }
+    
+    
+    void doOperations(String tType, String tValue, Map<String, ?> assyConfig) {
+        doOperations(tType, tValue, assyConfig, false);
+    }
+    
+    void doOperations(String tType, String tValue, Map<String, ?> assyConfig, boolean suppressUseToken) {
+        try {
+            Vantiq v = new Vantiq(TEST_SERVER, 1);
+            v.authenticate(SUB_USER, SUB_USER);
+            VantiqResponse resp;
+            TestFhirAssembly.installAssembly(v, assyConfig);
+            if (!suppressUseToken && (tType != null || tValue != null)) {
+                Map<String, String> params = new HashMap<>();
+                params.put("accessToken", tValue);
+                if (tType != null) {
+                    params.put("tokenType", tType);
+                }
+                resp = v.execute("com.vantiq.fhir.fhirService.useAccessToken",
+                                 params);
+                
+                assertTrue("Could not provide access token: " + resp.getErrors(), resp.isSuccess());
+            }
+            resp = v.execute("com.vantiq.fhir.fhirService.getSMARTConfiguration",
+                             Collections.emptyList());
+            assertTrue("Could not fetch SMART Config: " + resp.getErrors(), resp.isSuccess());
+            //noinspection unchecked
+            Map<String, String> foundHeaders =
+                    (Map<String, String>) mapper.readValue(new Gson().toJson(((JsonObject) resp.getBody())),
+                                                                                      Map.class);
+            log.debug("Found headers: {}", foundHeaders);
+            assertNotNull("Ho headers found", foundHeaders);
+            checkAuthHeader(tType, tValue, foundHeaders);
+            
+            Map<String, String> params = new HashMap<>();
+            
+            params.put("id", "doesn'tMatter");
+            params.put("type", "Patient");
+            resp = v.execute("com.vantiq.fhir.fhirService.read", params);
+            assertTrue("Could not read resource: " + resp.getErrors(), resp.isSuccess());
+            //noinspection unchecked
```

Comment:
The call above just returns whatever object is at that address.  the other calls are a bit different.